lib/europeana/api/record.rb

```diff
@@ -60,92 +51,46 @@ def params=(params = {})
       end
 
       ##
-      # Gets the URI for this Record request with parameters
+      # Gets the URL for this Record request
       #
       # @param [Hash{Symbol => Object}] options
       # @option options [Boolean] :ld (false)
       #   Request JSON-LD
-      # @return [URI]
-      def request_uri(options = {})
-        url = Europeana::API.url + "/record#{@id}.json"
-        url << 'ld' if options[:ld]
-        uri = URI.parse(url)
-        uri.query = params_with_authentication.to_query
-        uri
+      # @return [String]
+      def request_url(options = {})
+        (Europeana::API.url + "/record#{@id}.json").tap do |url|
+          url << 'ld' if options[:ld]
+        end
       end
 
+      alias_method :get, :execute_request
+
       ##
-      # Sends a request for this record to the API
+      # Examines the `success` and `error` fields of the response for failure
       #
-      # @param [Hash{Symbol => Object}] options
-      # @option options [Boolean] :ld (false)
-      #   Request JSON-LD
-      # @return [Hash]
-      # @raise [Europeana::Errors::ResponseError] if API response could not be
-      #   parsed as JSON
       # @raise [Europeana::Errors::RequestError] if API response has
       #   `success:false`
       # @raise [Europeana::Errors::RequestError] if API response has 404 status
       #   code
-      def get(options = {})
-        request = Request.new(request_uri(options))
-        response = request.execute
-        body = JSON.parse(response.body)
-        if (options[:ld] && !(200..299).include?(response.code.to_i)) || (!options[:ld] && !body['success'])
-          fail Errors::RequestError, (body.key?('error') ? body['error'] : response.code)
-        end
-        body
-      rescue JSON::ParserError
-        if response.code.to_i == 404
-          # Handle HTML 404 responses on malformed record ID, emulating API's
-          # JSON response.
-          raise Errors::RequestError, "Invalid record identifier: #{@id}"
-        else
-          raise Errors::ResponseError
-        end
-      end
-
-      ##
-      # Gets hierarchy data about this and related records from
-      # ancestor-self-siblings.json API method
-      #
-      # @note Proof of concept implementation for demo purposes.
-      # @todo Refactor if functionality to be retained.
-      def hierarchy(*args)
-        args = [:self] if args.blank?
-
-        data = {}
-        args.each do |method|
-          unless @hierarchy.key?(method)
-            @hierarchy[method] = hierarchical_data(method).select do |_k, v|
-              v.is_a?(Enumerable)
-            end
+      # @see Requestable#parse_response
+      def parse_response(response, options = {})
```

Comment:
Cyclomatic complexity for parse_response is too high. [8/6]<br>Assignment Branch Condition size for parse_response is too high. [20.25/15]<br>Method has too many lines. [13/10]<br>Perceived complexity for parse_response is too high. [9/7]